### DISCUSSION TITLE
[Bug] findAllActiveListingsByShop doesn't always return all active listings of the shop

### DISCUSSION THREAD

**Discussion by VasilVelikov** [etsy/open-api#873] [Q&A]
It seems like there is a bug in the endpoint **findAllActiveListingsByShop**. It doesn't always find all active listings in a shop.

Consider the call below:
`https://openapi.etsy.com/v3/application/shops/15490317/listings/active`

Currently, it returns 1030 active listings for the shop NAZCASPIRITBOUTIQUE. If you visit the shop on Etsy you'll see that it actually has 1032 active listings. If you make the call through the v2 API you will get 1032 as well.

There are many examples of shops like this. In all the cases I've found the missing listings in the API response are less than 1%. Also, I haven't seen a problematic shop with less than 100 listings.

Has anyone experienced this behavior? Is there a workaround?

Thanks

**Comment by VasilVelikov** [comment]
Hi,
I just tested the issue again and it still there. It can be reproduced in a stable way for many shops, not just for this one.
Can someone confirm that this is a bug?
Thanks!

**Reply by Gareth064** [reply]
in my test app, I get 1069 results for shop 15490317

![image](https://user-images.githubusercontent.com/22307223/221221460-3aaa9ef9-9069-48c1-b40e-26927724f06f.png)

**Comment by VasilVelikov** [comment]
Yes, the number of listings varies. At the moment:
- with API v3 I get 1069 listings.
- with API v2 I get 1071 listings
- opening the shop on Etsy (https://www.etsy.com/shop/NAZCASPIRITBOUTIQUE) I get 1071 listings.

The new API returns less listings than the old one and than the real number.

**Comment by Gareth064** [comment]
I'd email the Etsy developers to ask them to investigate. They don't use this GitHub for support issues anymore. This is now just a community support place.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by emergingdzns** [comment]
@VasilVelikov This came up before. The new api doesn't return listings that are flagged by the internal system for some reason. In the previous case the shop reporting it had two listings that were flagged as adult content but weren't appropriately categorized. I think there was also something to do with the listing availability based on the country requesting the data. Some listings aren't viewable in India for some reason.
